### PR TITLE
Bug/enrollment 404

### PR DESCRIPTION
**Pull request by ezturner**

### CODE REVIEW

File: uw_sws/tests/test_enrollment.py [hunk 1]
```diff
@@ -35,3 +36,15 @@ def test_javerage_major(self):
...
+            term = get_term_by_date(datetime.date(2014, 7, 13))
+            exception_raised = False
+            try:
+                enrollement = get_enrollment_by_regid_and_term('9136CCB8F66711D5BE060004AC494F31', term)
+            except DataFailureException as ex:
+                exception_raised = True
+                self.assertEquals(ex.url, "/student/v5/enrollment/2014,summer,9136CCB8F66711D5BE060004AC494F31.json")
+                self.assertEquals(ex.msg, "No enrollment found.")
+
+            self.assertTrue(exception_raised)
```
Comment (vegitron) [inline uw_sws/tests/test_enrollment.py]: Use https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises

File: uw_sws/enrollment.py [hunk 2]
```diff
@@ -49,11 +50,15 @@ def get_enrollment_by_regid_and_term(regid, term):
                                 term.year,
                                 term.quarter,
                                 regid)
-    return _json_to_enrollment(get_resource(url))
+    return _json_to_enrollment(get_resource(url), url)
```
Comment (vegitron) [inline uw_sws/enrollment.py]: get_resource raises a datafailureexception if the status isn't 200 - does the production sws return a proper 404 status code?  If so, our 404 resources should have a .http-headers file, and restclients will use that to load various parts of the response object.  here's an example: https://github.com/uw-it-aca/uw-restclients/blob/master/restclients/resources/bridge/file/api/admin/users/195.DELETE.http-headers

Comment (fanglinfang) [inline uw_sws/enrollment.py]: it should raise DataFailureException with a status 404 with or without  .http-headers file
How is this 404 resources differ from the regular one? why check in _json_to_enrollment ?

Comment (fanglinfang) [inline uw_sws/enrollment.py]: Ethan,
+ if "StatusCode" in json_data and json_data["StatusCode"] == "404":
+       raise DataFailureException(url, 404, json_data["StatusDescription"])
this piece is not right.  you may want to roll back the change in enrollment.py .

**Pull request closed without merging**